Cubical/Relation/Binary/Order/Woset/Base.agda

```diff
@@ -0,0 +1,206 @@
+{-# OPTIONS --safe #-}
+module Cubical.Relation.Binary.Order.Woset.Base where
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.Equiv
+open import Cubical.Foundations.Equiv.HalfAdjoint
+open import Cubical.Foundations.HLevels
+open import Cubical.Foundations.Isomorphism
+open import Cubical.Foundations.Univalence
+open import Cubical.Foundations.Transport
+open import Cubical.Foundations.SIP
+
+open import Cubical.HITs.PropositionalTruncation
+
+open import Cubical.Data.Sigma
+
+open import Cubical.Reflection.RecordEquiv
+open import Cubical.Reflection.StrictEquiv
+
+open import Cubical.Displayed.Base
+open import Cubical.Displayed.Auto
+open import Cubical.Displayed.Record
+open import Cubical.Displayed.Universe
+
+open import Cubical.Relation.Binary.Base
+open import Cubical.Relation.Binary.Extensionality
+
+open import Cubical.Induction.WellFounded
+
+open Iso
+open BinaryRelation
+
+
+private
+  variable
+    ℓ ℓ' ℓ'' ℓ₀ ℓ₀' ℓ₁ ℓ₁' ℓ₂ ℓ₂' : Level
+
+record IsWoset {A : Type ℓ} (_≺_ : A → A → Type ℓ') : Type (ℓ-max ℓ ℓ') where
+  no-eta-equality
+  constructor iswoset
+
+  field
+    is-set : isSet A
```

Comment:
This follows from prop-valued and weak extensionality.

Reply:
It does, but it helps to keep all of these orderings consistent and require the basics that they're prop-valued and over sets. For instance, posets require that A is a set, but that follows from prop-valued, reflexive, and antisymmetry.